x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts

```diff
@@ -4,35 +4,86 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Server } from 'hapi';
-import sinon from 'sinon';
-
-import { SavedObject } from 'src/legacy/server/saved_objects';
+import * as Rx from 'rxjs';
+import { SavedObject, SavedObjectsService } from 'src/legacy/server/saved_objects';
 import { Feature } from '../../../../xpack_main/types';
 import { convertSavedObjectToSpace } from '../../routes/lib';
 import { initSpacesOnPostAuthRequestInterceptor } from './on_post_auth_interceptor';
 import { initSpacesOnRequestInterceptor } from './on_request_interceptor';
-
+import { SpacesService } from '../../new_platform/spaces_service';
+import { SecurityPlugin } from '../../../../security';
+import { SpacesAuditLogger } from '../audit_logger';
+import { SpacesServiceSetup } from '../../new_platform/spaces_service/spaces_service';
+import { elasticsearchServiceMock, httpServiceMock } from '../../../../../../src/core/server/mocks';
+import * as kbnTestServer from '../../../../../../src/test_utils/kbn_server';
+import { HttpServiceSetup } from 'src/core/server';
+import { KibanaConfig, Server } from 'src/legacy/server/kbn_server';
+import { XPackMainPlugin } from '../../../../xpack_main/xpack_main';
+import { parse } from 'url';
+
+// TODO: re-implement on NP
 describe('onPostAuthRequestInterceptor', () => {
-  const sandbox = sinon.sandbox.create();
-  const teardowns: Array<() => void> = [];
   const headers = {
     authorization: 'foo',
   };
-  let server: any;
   let request: any;
+  let spacesService: SpacesServiceSetup;
 
   const serverBasePath = '/';
   const defaultRoute = '/app/custom-app';
 
+  let root: ReturnType<typeof kbnTestServer.createRoot>;
+
+  function initKbnServer(http: HttpServiceSetup) {
+    const kbnServer = kbnTestServer.getKbnServer(root);
+
+    kbnServer.server.route([
+      {
+        method: 'GET',
+        path: '/foo',
+        handler: (req: any) => {
+          return { path: req.path, basePath: http.getBasePathFor(req) };
```

Comment:
I changed Api slightly in https://github.com/elastic/kibana/pull/38237, could you update to `http.basePath.get(req)`?

Reply:
done!